lib/transaction.js

```diff
@@ -19,9 +21,9 @@ class Transaction {
    * @param {string} [options.deferrable] Sets the constraints to be deferred or immediately checked. PostgreSQL only
    */
   constructor(sequelize, options) {
+    this.hooks = new Hooks();
```

Comment:
Wont this means, now you can do `transaction.hook.add('beforeCreate')` only to find it won't work, as only `afterCommit` hooks is invoked in this module.

We should find a way to throw with `.add` method.

As I mentioned in another comment, if we go with `<object>.hooks.<hookName>` approach this can be avoided

Reply:
That is a good point, I tried to avoid this scenario with strict typings but not everyone uses that 😞 , however my comment https://github.com/sequelize/sequelize/pull/12160#discussion_r415230504 still applies, adding methods to hooks for all possible hooks would be misleading and using inheritance would lead to a lot of bloaty code.

Example:

```
class ModelHooks extends Hooks {
  beforeCreate(a, b, c): Promise<void> {
    return this.run(a, b, c);
  // A lot of code
}

class SequelizeHooks extends ModelHooks {
  // ...
}
```

I wanted to avoid making hooks` specific to their consumers but I guess this would work, I'd make the `run` method private tho to avoid too much API surface and potential misuse.